### ISSUE TITLE
Support OpenMP nest_lock

### ISSUE THREAD

**Issue by BradSwain** [coderrect-inc/OpenRace#197]
See the specification for [`omp_nest_lock`][1].

Should be similar to the normal omp lock, only this one can be acquired by the same thread multiple times.
The lockset analysis may need to be updated to handle this behavior.

See DataRaceBench cases 118 and 119 for examples (note they also contain section, but we should already support section).

[1]: https://www.openmp.org/spec-html/5.0/openmpsu157.html#x195-9410003.3.4

**Comment by bz-liu** [comment]
A nested lock can be locked several times. It doesn't unlock until you have unset it (`omp_unset_nest_lock`) as many times as the number of calls to `omp_set_nest_lock`. So, we should handle it as normal: push/pop when we see lock/unlock IR instructions, then for a specific nested lock, its number of locks and unlocks matches at the end and the lockset is correct (assume the code is correctly written). 

The current implementation uses `std::set` to store the lockset, this can early release a lock:
```cpp
..
omp_set_nest_lock // insert a new element: lock
omp_set_nest_lock // insert but no new element, return
x++
omp_unset_nest_lock // remove lock -> early realse
x++;  // no protection due to early realse -> report FP
omp_unset_nest_lock // nothing to release -> not match
...
```
To avoid this, now uses `std:: unordered_multiset`, seems faster than `std::multiset`. So the above example will be like: 
```cpp
..
omp_set_nest_lock // insert a new element: lock
omp_set_nest_lock // insert the same element: lock -> now we have {lock, lock}
x++
omp_unset_nest_lock // remove lock -> now we have {lock}
x++;  // still protected
omp_unset_nest_lock // remove lock -> now we have { } -> matched
...
```

**Comment by BradSwain** [comment]
closed by #223

**Comment by BradSwain** [comment]
closed by #223
